filebeat/input/filestream/input.go

```diff
@@ -163,7 +164,11 @@ func (inp *filestream) Run(
 	})
 	defer streamCancel()
 
-	return inp.readFromSource(ctx, log, r, fs.newPath, state, publisher, metrics)
+	if err := inp.readFromSource(ctx, log, r, fs.newPath, state, publisher, metrics); err != nil {
+		ctx.UpdateStatus(status.Degraded, fmt.Sprintf("error while reading from source: %v", err))
```

Comment:
Sorry for the late comment, if this exits with a transient error, what is setting the input state back to healthy?

In general the thing I am watching out for in the status reporting PRs situations where we permanently mark inputs as degraded or failed after the error condition has cleared.

Reply:
@cmacknz I understand your concern. 
I did some digging and turns out that `readFromSource` only throws an error when it calls `Publish()`
https://github.com/elastic/beats/blob/d87d15b406b0162989955f3f43053ff77192d415/filebeat/input/filestream/input.go#L394-L397
The above error is return when the context is cancelled. 
https://github.com/elastic/beats/blob/d87d15b406b0162989955f3f43053ff77192d415/filebeat/input/filestream/internal/input-logfile/publish.go#L91-L97.

For other cases (EOF, reader was closed etc.), we just logs the error and return `nil`. So, we will not mark the input as degraded, as they are transient in nature.

Reply:
@cmacknz maybe we don't need this particular `ctx.UpdateStatus`? I don't think we need to mark input as degraded when the context is cancelled (which should ideally happen during beat shutdown).

I overlooked this particular `ctx.UpdateStatus`.